### PR TITLE
fix(shim): Add 'Get-CommandPath()' to find git

### PR DESCRIPTION
**Pull request by rashil2000**
#### Description

Without `scoop-which`, the shimming function was picking up the ~\scoop\shims directory as source. This corrects it.

#### Motivation and Context

Closes https://github.com/ScoopInstaller/Main/issues/3562

#### Checklist:

- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have updated the tests accordingly.

### CODE REVIEW

**Comment by chawyehsu** [comment]
This is a bad implementation. To keep the libcore to be decoupled, it should never call other modules. An abstraction of extracting `which` from `scoop-which` to a reusable function would be better. Put it into libcore so functions elsewhere may use it when they need.

**Comment by niheaven** [comment]
I prefer a two-step check for this function: one for scoop-installed git, then a gcm one. scoop installed git could be checked by `appdir` and if not found, fall back to gcm.

**Comment by rashil2000** [comment]
Checking for scoop installed git using `appdir` is not feasible, plenty of packages provide the git.exe executable - git, git-with-openssh, git-without-openssh, deepgit, mingit, mingit-busybox etc. We can't put conditions for all of them.

**Comment by niheaven** [comment]
Hmm, so let's extract the main function of `scoop-which` and add it to `core.ps1` and try to avoid using `scoop-which` in such a situation IMO. That's the best way.

File: lib/core.ps1 [hunk 1]
```diff
@@ -318,6 +318,39 @@ function Get-HelperPath {
...
+    begin {
+        $userShims = Convert-Path (shimdir $false)
+        $globalShims = fullpath (shimdir $true) # don't resolve: may not exist
+    }
+
+    process {
+        try {
+            $comm = Get-Command $Command -ErrorAction Stop
+        } catch {
+            abort "'$Command' not found" 3
```
[the remark this answers is not in the document]
Comment (niheaven) [reply lib/core.ps1]: Changed, any idea?

[the remark this answers is not in the document]
Comment (rashil2000) [reply lib/core.ps1]: Yes, LGTM

**Review by niheaven — APPROVED** [review]

File: lib/core.ps1 [hunk 2]
```diff
@@ -318,6 +318,39 @@ function Get-HelperPath {
...
+    }
+
+    process {
+        try {
+            $comm = Get-Command $Command -ErrorAction Stop
+        } catch {
+            return $null
+        }
+        $commandPath = if ($comm.Path -like "$userShims*" -or $comm.Path -like "$globalShims*") {
+            Get-ShimTarget ($comm.Path -replace '\.exe$', '.shim')
```
Comment (Cologler) [inline lib/core.ps1:342]: If a bin includes both `*.ps1` and `*.exe` shims (like aria2c):

```
> gcm aria2c -all
...\shims\aria2c.ps1
...\shims\aria2c.exe
```

The command `scoop which aria2c` show `'aria2c' not found / not a scoop shim.`.

Comment (niheaven) [reply lib/core.ps1:342]: The new shim system should use only `.exe` shims, and `.ps1` shims are only used for `.cmd` or bash script.

Try remove all shims except `scoop.*` and `scoop reset *`, and this should be fixed.

**Pull request merged**